assets/javascripts/discourse/widgets/discourse-reactions-counter.js.es6

```diff
@@ -10,14 +12,77 @@ export default createWidget("discourse-reactions-counter", {
 
   buildId: attrs => `discourse-reactions-counter-${attrs.post.id}`,
 
+  init() {
+    this.subscribe();
+  },
+
+  unsubscribe() {
+    if (!this.attrs.post.id) {
+      return;
+    }
+    MessageBus.unsubscribe(`/post/${this.attrs.post.id}`);
+  },
+
+  subscribe() {
+    this.unsubscribe();
+
+    MessageBus.subscribe(
+      `/post/${this.attrs.post.id}`,
+      (data) => {
+        if(this.state[data.type].length) {
+          this.getUsers(data.type);
+        }
+      }
+    );
+  },
+
   defaultState() {
-    return {
-      statePanelExpanded: false
-    };
+    const state = {};
+
+    this.siteSettings.discourse_reactions_enabled_reactions.split('|').forEach((item) => {
+      state[item] = [];
+    });
+
+    state[this.siteSettings.discourse_reactions_reaction_for_like] = [];
+    state.statePanelExpanded = false;
+    state.postId = null;
+    state.reactionValue = null;
+    state.postIds = [];
+
+    return state;
+  },
+
+  getUsers(reactionValue = null) {
```

Comment:
do we need the`= null` here ? what does it give us compared to just `getUsers(reactionValue)` ?

Reply:
When we don't user null here and call the function without parameter, let's say `getUsers()` then `reactionValue` is undefined and we always need to check a condition `reactionValue === undefined` rather than `!reactionValue`

Reply:
why ?

![Capture d’écran 2021-03-08 à 15 01 03](https://user-images.githubusercontent.com/339945/110331236-2818fd80-801f-11eb-8024-059f4d097558.png)

Reply:
My bad, I got some misconceptions